ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -305,22 +911,49 @@ class Pagination {
      * Reflow the pagination container.
      */
     public reflow(): void {
-        if (!this.cell) {
+        const position = this.options.position;
+
+        if (!this.paginationContainer) {
             return;
         }
 
-        this.cell.style.width = this.grid.tableElement?.offsetWidth + 'px';
+        if (position === 'footer') {
+            // Set the width to match the table width
+            this.paginationContainer.style.width =
+                this.grid.tableElement?.offsetWidth + 'px';
+            return;
+        }
     }
 
     /**
-     * The pagination will be removed from the container.
+     * Destroy the pagination instance.
      */
     public destroy(): void {
-        this.row?.remove();
+        const position = this.options.position;
+
+        if (position === 'footer') {
+            // For footer position, remove the entire tfoot element.
+            this.paginationContainer?.parentElement?.parentElement?.remove();
+        } else if (typeof position === 'string' && position.startsWith('#')) {
+            // Don't remove the custom container itself
+            this.contentWrapper?.remove();
```

Comment:
I am pretty sure this will also destroy a custom container, since we set
`contentWrapper = customContainer` in `renderCustomContainer`
so this will end up removing the user created container